integration/fuse/src/main/java/alluxio/fuse/options/FuseOptions.java

```diff
@@ -59,34 +59,48 @@ public static FuseOptions create(AlluxioConfiguration conf, boolean updateCheckE
    * Creates the FUSE options.
    *
    * @param conf alluxio configuration
-   * @param options the file system options
+   * @param fileSystemOptions the file system options
    * @param updateCheckEnabled whether to enable update check
    * @return the file system options
    */
   public static FuseOptions create(AlluxioConfiguration conf,
-      FileSystemOptions options, boolean updateCheckEnabled) {
-    return new FuseOptions(options, constructFuseMountOptions(conf), updateCheckEnabled);
-  }
-
-  private static Set<String> constructFuseMountOptions(AlluxioConfiguration conf) {
-    Set<String> options = conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS)
+      FileSystemOptions fileSystemOptions, boolean updateCheckEnabled) {
+    Set<String> mountOptions = conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS)
         .stream().filter(a -> !a.isEmpty()).collect(Collectors.toSet());
-    if (AlluxioFuseUtils.getLibfuseVersion(conf) == LibfuseVersion.VERSION_2) {
+    LibfuseVersion version = AlluxioFuseUtils.getLibfuseVersion(conf);
+    if (!conf.getBoolean(PropertyKey.FUSE_JNIFUSE_ENABLED)
+        && version == LibfuseVersion.VERSION_3) {
+      throw new InvalidArgumentRuntimeException("Cannot use JNR-FUSE with libfuse 3");
+    }
+    if (version == LibfuseVersion.VERSION_2) {
       // Without option big_write, the kernel limits a single writing request to 4k.
       // With option big_write, maximum of a single writing request is 128k.
       // See https://github.com/libfuse/libfuse/blob/fuse_2_9_3/ChangeLog#L655-L659,
       // and https://github.com/torvalds/linux/commit/78bb6cb9a890d3d50ca3b02fce9223d3e734ab9b.
       // Libfuse3 dropped this option because it's default
       String bigWritesOptions = "big_writes";
-      options.add(bigWritesOptions);
-      LOG.info("Added fuse mount option {} to enlarge single write request size", bigWritesOptions);
-    }
-    if (!conf.getBoolean(PropertyKey.FUSE_JNIFUSE_ENABLED)) {
-      String directIOOptions = "direct_io";
-      options.add(directIOOptions);
-      LOG.info("Added fuse mount option {} for JNR FUSE", directIOOptions);
+      if (mountOptions.add(bigWritesOptions)) {
+        LOG.info("Added fuse mount option {} to enlarge single write request size",
+            bigWritesOptions);
+      }
+      if (!conf.getBoolean(PropertyKey.FUSE_JNIFUSE_ENABLED)) {
+        String directIOOptions = "direct_io";
+        if (mountOptions.add(directIOOptions)) {
+          LOG.info("Added fuse mount option {} for JNR FUSE", directIOOptions);
```

Comment:
We should also add direct_io for jnifuse too right?

Reply:
JNIFuse can support options other than direct_io.
especially for FUSE SDK, direct_io may not be favored

Reply:
tested -o max_idle_threads=<number> mount option works in libfuse 3. I don't think MAX_IDLE_THREADS env variable works though

Reply:
If we don't have direct_io, k8s will be unhappy? Isn't that the reason we added it?

The `MAX_IDLE_THREADS` env var doesn't work is expected. It's the hack we did on libfuse2.

Reply:
Leave the direct_io for user choice. Even if K8S is unhappy about the kernel_cache, many of our users still use kernel_cache because the much better performance in K8S

Reply:
Sounds good to me